include/ua_plugin_securitypolicy.h

```diff
@@ -0,0 +1,424 @@
+                                              const void *channelContext,
+                                              const UA_ByteString *certificate);
+
+    /**
+     * \brief Gets the signature size that depends on the remote public key.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param contextData the context to retrieve data from.
+     * \return the size of the remote asymmetric signature. Returns 0 if no remote certificate
+     *                     was set previousely.
+     */
+    size_t (*const getRemoteAsymSignatureSize)(const UA_SecurityPolicy *securityPolicy,
+                                               const void *contextData);
+
+    /**
+     * \brief Gets the plaintext block size that depends on the remote public key.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param contextData the context to retrieve data from.
+     * \return the size of the plain text block size when encrypting with the remote public key.
+     *         Returns 0 as long as no remote certificate was set previousely.
+     */
+    size_t (*const getRemoteAsymPlainTextBlockSize)(const UA_SecurityPolicy *securityPolicy,
+                                                    const void *contextData);
+
+    /**
+     * Gets the number of bytes that are needed by the encryption function in addition to the length of the plaintext message.
+     * This is needed, since most rsa encryption methods have their own padding mechanism included. This makes the encrypted
+     * message larger than the plainText, so we need to have enough room in the buffer for the overhead.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param contextData the retrieve data from.
+     * \param maxEncryptionLength the maximum number of bytes that the data to encrypt can be.
+     */
+    size_t (*const getRemoteAsymEncryptionBufferLengthOverhead)(const UA_SecurityPolicy *securityPolicy,
+                                                                const void *contextData,
+                                                                size_t maxEncryptionLength);
+};
+
+
+/**
+ * \brief This struct describes a security policy. All functions and modules need to be implemented.
+ */
+struct UA_SecurityPolicy {
+    /* The policy uri that identifies the implemented algorithms */
+    const UA_ByteString policyUri;
+
+    const UA_SecurityPolicyAsymmetricModule asymmetricModule;
+    const UA_SecurityPolicySymmetricModule symmetricModule;
+
+    const UA_Endpoint_SecurityContext endpointContext;
+    const UA_Channel_SecurityContext channelContext;
+
+    UA_Logger logger;
+};
+
+/**
+ * Holds an endpoint description and the corresponding security policy
+ * Also holds the context for the endpoint.
+ */
+typedef struct {

```

Comment:
(I assume you meant to post this comment below)
You're right. Wouldn't it be better though, to make it, so that we only have this defined once (use the struct here as well)
If we wouldn't use the struct, that would mean passing a size_t pointer and the array pointer.